kernel/vmbytegen.ml

```diff
@@ -875,10 +875,18 @@ let dump_bytecodes init code fvs =
 
 let compile ?universes:(universes=(0,0)) env sigma c =
   Label.reset_label_counter ();
+  let lam = lambda_of_constr env sigma c in
+  let params, body = decompose_Llam lam in
+  let arity = Array.length params in
+  let mask =
+    let rels = Genlambda.free_rels body in
+    let init i = if Int.Set.mem (arity - i) rels then true else false in
+    let mask = List.init arity init in
+    if List.for_all (fun b -> b) mask then [] else mask
```

Comment:
Rather than dropping the whole list when it is fully true, it would be better to drop any fully true suffix of the list, assuming I correctly understand its semantics.

Reply:
Something like
```ocaml
    let rec aux i acc =
      if i < 0 then acc
      else if Int.Set.mem (arity - i) rels && acc <> [] then true :: acc
      else false :: acc in
    aux (arity - 1) []
```

Reply:
Indeed, this is also possible, will do.